workers/unity/Assets/Gdk/Core/Systems/SpatialOSReceiveSystem.cs

```diff
@@ -55,15 +59,46 @@ private void OnDisconnect(DisconnectOp op)
             view.Disconnect(op.Reason);
         }
 
+        private void OnAddComponent(AddComponentOp op)
+        {
+            componentCallbacks[op.Data.ComponentId].OnAddComponent(op);
+        }
+
+        private void OnRemoveComponent(RemoveComponentOp op)
+        {
+            componentCallbacks[op.ComponentId].OnRemoveComponent(op);
+        }
+
+        private void OnComponentUpdate(ComponentUpdateOp op)
+        {
+            componentCallbacks[op.Update.ComponentId].OnComponentUpdate(op);
+        }
+
+        private void OnAuthorityChange(AuthorityChangeOp op)
+        {
+            componentCallbacks[op.ComponentId].OnAuthorityChange(op);
+        }
+
         private void SetupDispatcherHandlers()
         {
+            dispatcher.OnDisconnect(OnDisconnect);
+
             dispatcher.OnAddEntity(OnAddEntity);
             dispatcher.OnRemoveEntity(OnRemoveEntity);
-            dispatcher.OnDisconnect(OnDisconnect);
+
             dispatcher.OnCriticalSection(op => { inCriticalSection = op.InCriticalSection; });
 
+            dispatcher.OnAddComponent(OnAddComponent);
+            dispatcher.OnRemoveComponent(OnRemoveComponent);
+            dispatcher.OnComponentUpdate(OnComponentUpdate);
+            dispatcher.OnAuthorityChange(OnAuthorityChange);
+
             foreach (var translationUnit in view.TranslationUnits.Values)
             {
+                if (translationUnit is IDispatcherCallbacks)
```

Comment:
This feels odd, dont all units have this interface?

Reply:
WorldCommandsTranslation....